docs/02.API-REFERENCE.md

```diff
@@ -1710,41 +1710,70 @@ jerry_get_error_type (const jerry_value_t value);
 - [jerry_create_error](#jerry_create_error)
 - [jerry_value_is_error](#jerry_value_is_error)
 
-## jerry_value_clear_error_flag
+## jerry_get_value_from_error
 
 **Summary**
 
-Clear both the error and abort flags.
+Get the value from an error.
+
+Many API functions cannot be called with an error value.
+This function extracts the API value from an error. The second argument defines
+whether the input error value must be released or not. If it is set to `true`,
+then a [`jerry_release_value`](#jerry_release_value) function will be called
+for the first argument, so the error value won't be available after the call of
+`jerry_get_value_from_error`. The second argument should be false if both error
+and its represented value are needed. The first argument is returned unchanged if it is not an error.
+
+*Note*: Returned value must be freed with [jerry_release_value](#jerry_release_value) when it
+is no longer needed.
 
 **Prototype**
 
 ```c
-void
-jerry_value_clear_error_flag (jerry_value_t *value_p);
+jerry_value_t
+jerry_get_value_from_error (jerry_value_t value, bool release)
 ```
 
-- `value_p` - pointer to an api value
+- `value` - error (api) value
+- `release` - raw boolean, defines whether input value must be released
+- return value - api value
 
-**Example**
+**Example 1**
 
 ```c
 {
-  jerry_value_t value;
+  jerry_value_t error;
   ... // create or acquire value
 
-  jerry_value_clear_error_flag (&value);
+  jerry_value_set_error_flag (&error);
+  jerry_value_t value = jerry_get_value_from_error (error, true);
+  // using the error variable after release is invalid.
 
   jerry_release_value (value);
 }
 ```
 
+**Example 2**
+
+```c
+{
+  jerry_value_t error;
+  ... // create or acquire value
+
+  jerry_value_set_error_flag (&error);
+  jerry_value_t value = jerry_get_value_from_error (error, false);
+
```

Comment:
`... // both 'error' and 'value' can be used and must be released when they are no longer needed`

Reply:
Thanks, updated the PR.